### PR TITLE
[WIP] Update maximum value of numberwidth to NUMBUFLEN (65)

### PR DESCRIPTION
**Pull request by shtanton**
This addresses issue #4722

The docs file is also updated but I'm not sure if it could be improved.

### CODE REVIEW

**Comment by shtanton** [comment]
I don't know how this failed the check because I barely changed anything, help would be appreciated

**Comment by blueyed** [comment]
> From test_options.vim:
> Found errors in Test_set_errors():
> function RunTheTest[37]..Test_set_errors line 5: command did not fail: set numberwidth=11

(run it via `TEST_FILE=test_options.res make oldtest`)

Also "lint" is failing - have not checked that one.
See `make lint`.

**Comment by shtanton** [comment]
As far as I can tell, I fixed the unit test with one commit, and only the linter wasn't working.

I then fixed that and somehow the unit tests started failing again, and it seems to be a part of the code that I haven't touched. My commit only changed a comment so I have no idea how this happened.

Thanks for the help

**Comment by blueyed** [comment]
> I then fixed that and somehow the unit tests started failing again

Likely a flaky test then.

Yes:
```
TUI FocusGained/FocusLost in cmdline-mode does NOT :echo: [1m[31mERR[0m[0m
test/functional/ui/screen.lua:564: expected screen state to be unchanged

stack traceback:
	test/functional/ui/screen.lua:564: in function '_wait'
	test/functional/ui/screen.lua:366: in function 'expect'
	.../neovim/neovim/test/functional/terminal/tui_spec.lua:389: in function <.../neovim/neovim/test/functional/terminal/tui_spec.lua:376>
```

You can ignore that.

**Comment by blueyed** [comment]
Thanks for this.

I think this would make sense in Vim already, doesn't it?

**Comment by shtanton** [comment]
Yes, I can make a similar PR for vim if you think that's a good idea

**Comment by blueyed** [comment]
@shtanton 
Yes, it would be good to have this fixed in Vim (create an issue with patch, or PR with them), and then the patch would also be imported here.  This helps keeping the diff small for shared code etc.
It also avoids having different behavior in case Bram/Vim decides to use something else.
Thanks!

**Comment by shtanton** [comment]
Not sure if I should post this here or create my pull request on vim, but in (src/screen.c line 3170)[https://github.com/vim/vim/blob/master/src/screen.c#L3170] the string for the line number bit is set to 18 bytes long, the same is actually true for (neovim)[https://github.com/neovim/neovim/blob/master/src/nvim/screen.c#L2064].

Because of this the same change in vim didn't work until I increased this value.

My questions are:
1. What should I increase the value to, does 18 have any significance?
2. Why did this work in neovim?

Thanks for helping me with this

**Comment by shtanton** [comment]
I thought about it a bit and my guess is that when I tested it in neovim the bytes after that string were allocated to neovim by my os so it just overwrote some other variables probably, but kept working. It did work a couple of times in vim when it shouldn't have so this makes sense to me.

To prevent this I'll add tests that check setting nuw to 65, as I think the only tests at the moment check the errors but not the successful side of the boundary.

File: src/nvim/vim.h [hunk 1]
```diff
@@ -23,7 +23,11 @@
 
 #include "nvim/os/os_defs.h"       // bring lots of system header files
 
-/// length of a buffer to store a number in ASCII (64 bits binary + NUL)
+// length of a buffer to store a number in ASCII (64 bits binary + NUL)
+// if this changes the max numberwidth tests in
+// test/functional/options/num_options_spec.lua and
+// src/nvim/testdir/test_options.vim
+// should be updated
```
Comment (justinmk) [inline src/nvim/vim.h]: The tests will fail and that should make it obvious that they need to be updated, so we can avoid comments like this.

Comment (shtanton) [reply src/nvim/vim.h]: Cool, I'll leave that out when I do the vim PR

**Comment by shtanton** [comment]
I'm closing this and I'm gonna open a pull request for this on vim at some point

**Comment by shtanton** [comment]
[Here](https://github.com/vim/vim/pull/4584) is the vim PR

**Pull request closed without merging**